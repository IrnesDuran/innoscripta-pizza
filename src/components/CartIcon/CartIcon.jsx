import React from 'react';
import './CartIcon.styles.scss';
import  { connect } from 'react-redux';


import { ReactComponent as ShoppingIcon } from '../../assets/shopping-bag.svg';

import { toggleCartHidden } from '../../redux/cart/cart.actions'



const CartIcon =(props) => (
    <div className='cart-icon' onClick={props.toggleCartHidden}>
        <ShoppingIcon className='shopping-icon' />
        <span className='item-count'>{props.itemCount}</span>
    </div>
);


const mapStateToProps = state => ({
    itemCount: state.cart.cartItems.length
  });
  

const mapDispatchToProps = dispatch => ({
    toggleCartHidden: () => dispatch(toggleCartHidden())

});


export default connect(mapStateToProps, mapDispatchToProps)(CartIcon);

