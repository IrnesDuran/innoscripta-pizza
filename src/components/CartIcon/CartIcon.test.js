import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });


import {CartIcon} from './CartIcon';



describe('Testing CartIcon value',()=>{

    it('should test CartIcon snapshot',()=>{
        expect(shallow(<CartIcon/>)).toMatchSnapshot();

    });

    it('should test if cart icon value is 0',()=>{
        const wrapper = shallow(<CartIcon/>);
        const count =wrapper.find('span.item-count');
        expect(count.text()).toBe("");
    });


    it('should test if cart icon value is incramented',()=>{
        const mockProps = {
            itemCount: 1,
            toggleCartHidden:jest.fn()
      };
        const wrapper = shallow(<CartIcon {...mockProps}/>);
        expect(wrapper.text()).toBe("1");
    });

});

