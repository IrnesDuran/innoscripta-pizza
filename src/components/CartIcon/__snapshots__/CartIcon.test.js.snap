// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing CartIcon value should test CartIcon snapshot 1`] = `ShallowWrapper {}`;
